sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -61,5 +65,24 @@ public RequestFailedException(int status, string message, string? errorCode, Exc
             Status = status;
             ErrorCode = errorCode;
         }
+
+        /// <inheritdoc />
+        protected RequestFailedException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+            Status = info.GetInt32(nameof(Status));
```

Comment:
What will this do when the information is not in the stream?

Reply:
It will throw but this class owns both serialization and deserialization logic so it would never happen in a normal usecase.